bot/role_management/role_management.py

```diff
@@ -131,6 +130,203 @@ async def module_role_error(self, ctx: commands.Context, error: commands.Command
+        """
+        if message.channel.id != self.ch_role.id:
+            await ctx.send(f":information_source: Reaction-Roles können nur für Nachrichten im Kanal "
+                           f"{self.ch_role.mention} erstellt werden.")
+            return
+        if emoji in [reaction.emoji for reaction in message.reactions]:
+            await ctx.send(":x: Für den angegebenen Emoji existiert bereits eine Reaction-Role.")
+            return
+
+        self._db_connector.add_reaction_role(message.id, emoji, role.id)
+        await message.add_reaction(emoji)
+        log.info("A reaction role has been added to the message with id %s.", message.id)
+
+        await ctx.send(":white_check_mark: Die Reaction-Role wurde erfolgreich erstellt.")
+
+    @reaction_role.command(name="remove")
+    @commands.is_owner()
+    @command_log
+    async def remove_reaction_role(self, ctx: commands.Context, message: discord.Message, emoji: str):
+        """Command Handler for the subcommand `remove` of the `reactionrole` command.
+
+        Removes the reaction from a specified message and deletes the corresponding database entry of the reaction role.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            message (discord.Message): The message from which the reaction role should be removed.
+            emoji (str): The emoji of the reaction.
+        """
+        if message.channel.id != self.ch_role.id:
+            await ctx.send(f":information_source: Nachrichten außerhalb des Kanals {self.ch_role.mention} können keine "
+                           f"Reaction-Roles besitzen.")
+            return
+        if emoji not in [reaction.emoji for reaction in message.reactions]:
+            await ctx.send(":x: Für den angegebenen Emoji existiert leider keine Reaction-Role.")
+            return
+
+        self._db_connector.remove_reaction_role(message.id, emoji)
+        await message.clear_reaction(emoji)
+        log.info("A reaction role has been removed from the message with id %s.", message.id)
+
+        if len(message.reactions) == 1:
+            self._db_connector.remove_reaction_role_group(message.id)
+
+        await ctx.send(":white_check_mark: Die Reaction-Role wurde erfolgreich entfernt.")
+
+    @reaction_role.command(name="clear")
+    @commands.is_owner()
+    @command_log
+    async def clear_reaction_roles(self, ctx: commands.Context, message: discord.Message):
+        """Command Handler for the subcommand `clear` of the `reactionrole` command.
+
+        Removes all the reaction from a specified message and deletes all the corresponding database entry of the
+        reaction roles.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            message (discord.Message): The message from which the reaction roles should be removed.
+        """
+        if message.channel.id != self.ch_role.id:
```

Comment:
You could pull this out of the submethods and make it a check or something like that.

Reply:
Yes I could but I'm unsure if this would be a good idea. As of now we have this one-liner four times in our code. Creating a check and calling it would result in at least two lines of code more than we already have. I think it would make sense if we would need the same check more often.

Reply:
Why two lines more?
You'd have this code ( 3lines) only once and put an annotation (?) onto the methods right=

Reply:
Yeah if you would use it as an annotation. But then I wouldn't be able to give the user feedback this way. Like sending a message in the channel because the command simply wouldn't execute.